Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -2116,6 +2116,12 @@ internal struct AvailableDevice
         internal IInputRuntime m_Runtime;
         internal InputMetrics m_Metrics;
         internal InputSettings m_Settings;
+
+        // Extract as booleans (from m_Settings) because feature check is in the hot path
+        internal bool m_OptimizedControlsFeatureEnabled;
+        internal bool m_ReadValueCachingFeatureEnabled;
+        internal bool m_ParanoidReadValueCachingChecksEnabled;
```

Comment:
Could you please change these to private fields and add an internal read-only accessor property?

When refactoring for CoreCLR support, internal fields were very frustrating because I had to check the entire code base for any usages.

Reply:
This have been adressed now in https://github.com/Unity-Technologies/InputSystem/pull/1954/commits/c93e565d0b9260c0911f2af882871dd8551fe80a

Reply:
Regarding FWPM refactor I think this is straightforward since ApplySettings seem intact.